lib/p2p.go

```diff
@@ -281,6 +286,20 @@ func StartP2PInstance(argIP, argMac, argDev, argDirect, argHash, argDht, argKeyf
 	return p
 }
 
+func (p *PeerToPeer) attemptPortForward(port uint16, name string) error {
+	Log(Info, "Trying to forward port %d", port)
+	d, err := upnp.Discover()
+	if err != nil {
+		return err
+	}
+	err = d.Forward(port, "subutai-"+name)
+	if err != nil {
+		return err
+	}
+	Log(Info, "Port %d has been forwarded", port)
+	return nil
+}
+
 // Init will initialize PeerToPeer
```

Comment:
I guess this only works if the router has enabled UPnP correct?

Reply:
Correct.